### PR TITLE
feat(api,ui): interpolate matrix values

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/api/v2_workflow_run_engine.go [hunk 1]
```diff
@@ -726,15 +803,122 @@ func prepareRunJobs(_ context.Context, run sdk.V2WorkflowRun, wrEnqueue sdk.V2Wo
...
+			bts, _ := json.Marshal(rootJobContext)
+			var mapContexts map[string]interface{}
+			_ = json.Unmarshal(bts, &mapContexts) // error cannot happen here
+
+			ap := sdk.NewActionParser(mapContexts, sdk.DefaultFuncs)
+
+			for k, v := range jobDef.Strategy.Matrix {
 				keys = append(keys, k)
+
+				log.Warn(ctx, ">>>>>BOULB::::: %T", v)
```
Comment (fsamin) [inline engine/api/v2_workflow_run_engine.go:849]: nore sure

**Review by fsamin — CHANGES_REQUESTED** [review]

File: engine/api/v2_workflow_run_engine.go (hunk 1 above)
Comment (sguiheux) [inline engine/api/v2_workflow_run_engine.go:849]:
```suggestion
```

**Review by fsamin — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#21035.1 ✔ 
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#21035.1 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Pull request merged**